src/domains/integration/service/integration.service.impl.ts

```diff
@@ -65,120 +40,37 @@ export class IntegrationServiceImpl implements IntegrationService {
-    const integrator: UserDTO = await this.getProjectIntegrator(pendingProject.token);
-    const pendingMemberMails: string[] = (await this.pendingMemberMailsRepository.getByProjectId(pendingProject.id)).map((memberMail) => memberMail.email);
-    const organization: OrganizationDTO = await this.getOrganization(pendingProject.organizationId);
+    const { email: integratorEmail } = await this.getProjectIntegrator(pendingProject.token);
+    const authorizedMemberMails: string[] = (await this.pendingMemberMailsRepository.getByProjectId(pendingProject.id)).map((memberMail) => memberMail.email);
+    const { id: organizationId } = await this.getOrganization(pendingProject.organizationId);
     Logger.timeEnd(`Verifications`);
-    Logger.time('Members, Stages, Labels adaptation');
-    const projectData: ProjectDataDTO = await this.adapter.adaptProjectData({ providerProjectId: projectId, pmEmail: integrator.email, token: pendingProject.token, memberMails: pendingMemberMails });
-    Logger.timeEnd(`Members, Stages, Labels adaptation`);
-    await this.verifyPmExistence(projectData.members, integrator.email);
-    const memberRoles: UserRole[] = await this.assignRoles(projectData.members, integrator.email);
-
-    const project: ProjectDTO = await db.$transaction(
-      async (db: Omit<PrismaClient, ITXClientDenyList>): Promise<ProjectDTO> => {
-        const projRep: ProjectRepositoryImpl = new ProjectRepositoryImpl(db);
-        const newProject: ProjectDTO = await projRep.create(projectData.projectName, projectId, organization.id, projectData.image ?? null);
-        Logger.time('Members integration');
-        await this.integrateMembers({
-          memberRoles,
-          projectId: newProject.id,
-          emitterId: integrator.id,
-          acceptedUsers: pendingMemberMails,
-          db,
-        });
-        Logger.timeEnd(`Members integration`);
-        Logger.time('Stages Integration');
-        await this.integrateStages({ projectId: newProject.id, stages: projectData.stages, db });
-        Logger.timeEnd(`Stages Integration`);
-        Logger.time('Labels integration');
-        await this.integrateLabels({ projectId: newProject.id, labels: projectData.labels, db });
-        Logger.timeEnd(`Labels integration`);
-        Logger.time('Issues integration');
-        await this.integrateIssues({ projectId: newProject.id, issues: projectData.issues, db });
-        Logger.timeEnd(`Issues integration`);
-
-        return newProject;
-      },
-      {
-        maxWait: 1000, // default: 2000
-        timeout: 20000, // default: 5000
-      }
-    );
+    Logger.time('Provider project data adaptation adaptation');
+    const projectData: ProjectDataDTO = await this.adapter.adaptProjectData({ providerProjectId: projectId, pmEmail: integratorEmail, token: pendingProject.token });
+    Logger.timeEnd(`Provider project data adaptation adaptation`);
+    await this.verifyPmExistence(projectData.members, integratorEmail);
```

Comment:
Please add return type explicitly, it is easier to follow the flow while checking code

Reply:
done